src/state/slices/addressBookSlice/selectors.ts

```diff
@@ -121,6 +123,13 @@ export const selectInternalAccountIdByAddress = createDeepEqualOutputSelector(
       return accountId ?? null
     }
 
+    // For UTXO chains, use progressive address derivation to find matches
+    // This intelligently searches in batches (20, 50, 100) to handle edge cases
+    // where users have many transactions and the address is beyond the standard gap limit
```

Comment:
q1: where does this logic live?
q2: This whole logic does assume that said request *was* indeed fired and react-query has cache for it right? Seems like a very optimistic solution with potentially many edge cases

Reply:
Removed the comment in [dae8b45](https://github.com/shapeshift/web/pull/10928/commits/dae8b45dc9e90ec32cc489375fee088726e2a112)

Which kind of edge cases would you think about? If we don't have the data, we can't match accounts as anyway we don't have the UTXO accounts, so you probably can't even access the send feature for UTXOs at this point, also see previous comments for more informations on my choice